.travis.yml

```diff
@@ -33,10 +33,24 @@ jobs:
     - stage: Build test
       language: minimal
       before_script:
-        - LATEST_AZCOPY="$(curl -sSLo- https://api.github.com/repos/Azure/azure-storage-azcopy/releases | jq -r '.[0].tag_name' | sed 's/^v//g')"
-        - test -n "$LATEST_AZCOPY"
-        - export LATEST_AZCOPY
+        - if [ "master" = "$SOURCE_BRANCH" ]; then
+            AZCOPY_VERSION=$(curl -sLo- https://api.github.com/repos/Azure/azure-storage-azcopy/releases | grep tag_name | head -n 1 | awk -F'"' '{print $4}' | sed 's/^v//g');
+          else
+            AZCOPY_VERSION="$SOURCE_BRANCH";
+          fi
```

Comment:
Looks like we have a problem here.

According to the docs: https://docs.travis-ci.com/user/environment-variables/, we should use `$TRAVIS_BRANCH`, right?

> - for push builds, or builds not triggered by a pull request, this is the name of the branch.
> - for builds triggered by a pull request this is the name of the branch targeted by the pull request.
> - for builds triggered by a tag, this is the same as the name of the tag (`TRAVIS_TAG`).

Using `AZCOPY_VERSION="$TRAVIS_BRANCH"` will still have problem for pull requests.

What about to check `$TRAVIS_TAG` first, assign it to `$AZCOPY_VERSION` when it's existed? When it doesn't exist, we use `curl` to determinate the version.

Reply:
@odidev this one is not changed since the last few review.